src/main/java/jdbchomework/dao/jdbc/AbstractDao.java

```diff
@@ -2,118 +2,116 @@
+    protected Connection connection;
+
+    public AbstractDao(Connection connection, String table, String column) {
+        this.connection = connection;
+        this.table = table;
+        this.column = column;
+    }
+
+    @Override
+    public boolean deleteById(long id) {
+        boolean result = false;
+        try (PreparedStatement statement = connection
+                .prepareStatement("DELETE FROM " + table + " WHERE " + column + " = ?;")) {
+            statement.setLong(1, id);
+            if (statement.executeUpdate() > 0) {
+                result = true;
+            }
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException(e);
+        }
+        return result;
     }
 
     public void add(T toAdd) {
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            session.save(toAdd);
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (PreparedStatement statement = connection.prepareStatement("INSERT  INTO " + table + " (name)VALUES (?);")) {
+            connection.setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
+            connection.setAutoCommit(false);
+            statement.setString(1, toAdd.getName());
+            connection.commit();
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
+        } finally {
+            try {
+                connection.setAutoCommit(true);
+            } catch (SQLException e) {
+                log.error("Cannot connect to DB", e);
+                throw new RuntimeException("Cannot connect to DB", e);
             }
-            e.printStackTrace();
         }
     }
 
     public List<T> getAll() {
         List<T> result = new ArrayList<>();
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            result = session.createQuery("FROM " + entity).list();
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (Statement statement = connection.createStatement()) {
+            String sql = "SELECT * FROM " + table;
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                T t = createEntity(resultSet);
+                result.add(t);
             }
-            log.error(CANNOT_CONNECT_TO_DB, e);
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)